packages/notebook/src/widget.ts

```diff
@@ -1578,28 +1578,40 @@ export class Notebook extends StaticNotebook {
   }
 
   /**
-   * Handle `contextmenu` event.
+   * Find the target of html mouse event and cell index containing this target.
+   *
+   * #### Notes
+   * Returned index is -1 if the cell is not found.
    */
-  private _evtContextMenuCapture(event: PointerEvent): void {
-    // Allow the event to propagate un-modified if the user
-    // is holding the shift-key (and probably requesting
-    // the native context menu).
-    if (event.shiftKey) {
-      return;
-    }
-    // `event.target` sometimes gives an orphaned node in Firefox 57, which
-    // can have `null` anywhere in its parent tree. If we fail to find a
-    // cell using `event.target`, try again using a target reconstructed from
-    // the position of the click event.
+  private _findEventTargetAndCell(event: MouseEvent): [HTMLElement, number] {
     let target = event.target as HTMLElement;
     let index = this._findCell(target);
     if (index === -1) {
+      // `event.target` sometimes gives an orphaned node in Firefox 57, which
+      // can have `null` anywhere in its parent line. If we fail to find a cell
+      // using `event.target`, try again using a target reconstructed from the
+      // position of the click event.
       target = document.elementFromPoint(
         event.clientX,
-        event.clientY
+        event.clientY,
       ) as HTMLElement;
       index = this._findCell(target);
     }
+    return [target, index];
+  }
+
+  /**
+   * Handle `contextmenu` event.
+   */
+  private _evtContextMenuCapture(event: PointerEvent): void {
+    // Allow the event to propagate un-modified if the user
+    // is holding the shift-key (and probably requesting
+    // the native context menu).
+    if (event.shiftKey) {
+      return;
+    }
+
+    const [target, index] = this._findEventTargetAndCell(event);
```

Comment:
The diff here is a little bit weird so I describe this change.

Here, I've taken some of the code from `_evtContextMenuCapture()` and separated it into a new function called `_findEventTargetAndCell()`.